webhooks-extension/src/components/WebhookDisplayTable/WebhookDisplayTable.js

```diff
@@ -0,0 +1,218 @@
+import React, { Component } from 'react';
+import { Link, Redirect } from 'react-router-dom'
+
+import './WebhookDisplayTable.scss';
+import Delete from '@carbon/icons-react/lib/delete/16';
+import AddAlt24 from '@carbon/icons-react/es/add--alt/16';
+import { getWebhooks } from '../../api'
+
+import {
+  Button,
+  DataTable,
+  TableSelectAll,
+  TableSelectRow,
+  TableToolbar,
+  TableToolbarContent,
+  TableBatchActions,
+  TableBatchAction,
+  TableToolbarSearch,
+  Loading,
+  InlineNotification,
+} from 'carbon-components-react';
+
+const {
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableBody,
+  TableCell,
+  TableHeader,
+} = DataTable;
+
+export class WebhookDisplayTable extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      error: null,
+      isLoaded: false,
+      webhooks: [],
+      showNotification: false,
+      notificationErrorMessage: '',
+      notificationStatus: '',
+      notificationStatusMsgShort: ''
+    };
+
+  }
+
+  async componentDidMount() {
+    let data;
+    try {
+      data = await getWebhooks();
+      this.setState({
+        isLoaded: true,
+        webhooks: data
+      });
+    } catch (error) {
+      error.response.text().then((text) => {
+        this.setState({
+          notificationErrorMessage: "Failure occured fetching webhooks, error returned from the REST endpoint was : " + text,
+          notificationStatus: 'error',
+          notificationStatusMsgShort: 'Error:',
+          showNotification: true,
+        });
+      });
+    }
+  }
+
+  handleSelectedRows = (rows) => {
+    console.log(rows);
+  }
+
+  formatCellContent(id, value) {
+    // Render the git repo as a clickable link
+    if (id.endsWith(":repository")) {
+      return <a href={value} target="_blank">{value}</a>
+    } else {
+      return value
+    }
+  }
+
+  render() {
+
+    if (this.state.isLoaded) {
+      if (!this.state.webhooks.length) {
+        return (
+          // There are no webhooks, so redirect to the create panel
+          <Redirect to={this.props.match.url + "/create"} />
+        )
+      } else {
+        // There are webhooks so display table
+        const headers = [
+          {
+            key: 'name',
+            header: 'Name',
+          },
+          {
+            key: 'repository',
+            header: 'Git Repository',
+          },
+          {
+            key: 'pipeline',
+            header: 'Pipeline',
+          },
+          {
+            key: 'namespace',
+            header: 'Namespace',
+          }
+        ];
```

Comment:
Are these keys and headers necessary when the header and key are the same, could you use just a list of header names?

Reply:
The header and key are subtly different ... there is capitalisation in the Header and in one case spacing ... "Git Repository" ..... the 'key' value is matched in the setting up of initialRows .... so im currently inclined to keep it as it is.